src/react-components/object-info-dialog.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import DialogContainer from "./dialog-container.js";
+import styles from "../assets/stylesheets/client-info-dialog.scss";
+import { FormattedMessage } from "react-intl";
+
+export default class ObjectInfoDialog extends Component {
+  static propTypes = {
+    scene: PropTypes.object,
+    el: PropTypes.object,
+    objectDisplayString: PropTypes.string,
+    onClose: PropTypes.func,
+    hubChannel: PropTypes.object
+  };
+
+  state = {
+    pinned: false
+  };
+
+  componentDidMount() {
+    this.updatePinnedState = this.updatePinnedState.bind(this);
+    this.pin = this.pin.bind(this);
+    this.unpin = this.unpin.bind(this);
+    this.props.scene.addEventListener("uninspect", this.props.onClose);
+    this.updatePinnedState();
+  }
+
+  updatePinnedState() {
+    this.setState({ pinned: this.props.el.components["networked"].data.persistent });
+  }
+
+  pin() {
+    if (!NAF.utils.isMine(this.props.el) && !NAF.utils.takeOwnership(this.props.el)) return;
+    this.props.el.emit("pinned", { el: this.props.el });
+    this.props.el.setAttribute("pinnable", "pinned", true);
+    this.updatePinnedState();
+  }
+
+  unpin() {
+    if (!NAF.utils.isMine(this.props.el) && !NAF.utils.takeOwnership(this.props.el)) return;
+    this.props.el.emit("unpinned", { el: this.props.el });
+    this.props.el.setAttribute("pinnable", "pinned", false);
+    this.updatePinnedState();
+  }
+
+  delete() {
+    const targetEl = this.props.el;
+
+    if (!NAF.utils.isMine(targetEl) && !NAF.utils.takeOwnership(targetEl)) return;
+
+    targetEl.setAttribute("animation__remove", {
+      property: "scale",
+      dur: 200,
+      to: { x: 0.01, y: 0.01, z: 0.01 },
+      easing: "easeInQuad"
+    });
+
+    targetEl.addEventListener("animationcomplete", () => {
+      this.props.scene.systems["hubs-systems"].cameraSystem.uninspect();
+      NAF.utils.takeOwnership(targetEl);
+      targetEl.parentNode.removeChild(targetEl);
```

Comment:
I think you might have a race condition here you might need to guard for: If someone else deletes this object before this animation completes,  targetEl.parentNode may not be set when this callback fires.

Reply:
Yes I think you're right. Networked messages are paused until I leave the inspect mode, so the likelihood of this happening is very low, but it IS technically possible to quickly hit "delete" and then "cancel" to uninspect (thereby leaving pause mode) before the animation completes. I'll fix this.